### ISSUE TITLE
3D point cloud visualization

### ISSUE THREAD

**Issue by pxing6** [canonical-capsules/canonical-capsules#11]
hi,weiwei.when i use canonical to train 3D point cloud ,but when i vision it by your visual script(./main.py --save_dir=gifs_plane3d --indim=3 -- --mode=vis --pt_file=logs/plane_dim3/checkpoint.pth), it look terrible ,can you give me a hand,whether something i did wrong
plane
![image](https://user-images.githubusercontent.com/84260397/173311903-bea0fe68-aa3c-45dc-8002-4375e185bcb7.png)
![image](https://user-images.githubusercontent.com/84260397/173311921-16fb62f0-b644-4985-bdb3-19f9b532edb3.png)
car
![image](https://user-images.githubusercontent.com/84260397/173312049-615febb0-8bbb-437b-9b76-f677f22b86a3.png)

**Comment by wsunid** [comment]
Hi, Seems the axes have a different unit -- because the plane looks deformed. BTW, have you checked the training losses -- e.g., chamfer distance? has the training converged?

**Comment by pxing6** [comment]
yes ,i have checked training losses,the training has converged.
![image](https://user-images.githubusercontent.com/84260397/173495986-b7d972ee-7db2-46d4-be9b-5f4e7cca37e3.png)
![image](https://user-images.githubusercontent.com/84260397/173496005-4dc8405b-1693-433a-8993-56c40063c097.png)
i am wondering if some thing wrong with my datadoader or some other mistakes.

**Comment by pxing6** [comment]
the data i use is AtlasNetV1(shapenetv1pointcloud 、 shapenetv1renderings and shapenetcorev2Normalized),and use your script to transform them into h5 file.

**Comment by wsunid** [comment]
Sorry for the late. If this loss is normal -- you might want to double-check with our evaluation pipeline, maybe the reconstruction is not just great for these samples. Do check other samples. And, BTW, we're using 1024 points, which might also give us a bad visualization.

**Comment by pxing6** [comment]
hi,weiwei.other samples visual is also awful.i think is my dataloader's problem.because the links of customShapeNet and ShapeNetRendering are invaild,i download the same data from AtlasnetV1.I transform V2 normalized data and V1pointcloud、render into h5 file.,Is there something wrong with this place?Do you think so?
![image](https://user-images.githubusercontent.com/84260397/174248863-607e3cbe-19d2-429f-82ab-dddc28b46928.png)

**Comment by wsunid** [comment]
Not very sure how you download the dataset. It has been a while -- I don't quite remember the details of the dataset. I guess the easiest way is to run the evaluation pipeline to double-check.

**Comment by pxing6** [comment]
thanks for your reply.Ok,I will try it.i follow your train steps,i can only see informations such as losses and epochs.Do the evaluation pipeline exist in your codes .

**Comment by wsunid** [comment]
yeah, I think so. 

https://github.com/canonical-capsules/canonical-capsules/blob/fd4a69c19073ab63bb1511703386dcd9eed9e021/network.py#L260

**Comment by pxing6** [comment]
i have checked it,the chamfer_error and main_metric are both low.
![image](https://user-images.githubusercontent.com/84260397/174718573-35be7d89-2f47-43f6-b4a5-e9a43b0b20e6.png)

**Comment by pxing6** [comment]
Could you offer customShapenet and ShapeNetRendering.Atlasnet author don't support the two datasets.So i use other datasets to reproduction your experiment,but 3D result isn't well
